### PR TITLE
[Update]: Add Tumblr provider 

### PR DESCRIPTION
**Pull request by mikah13**
Reference:

- [Naver](https://developers.naver.com/docs/login/api/api.md) 
- [Tumblr](https://www.tumblr.com/docs/en/api/v2)

### CODE REVIEW

**Comment by mikah13** [comment]
@pilcrowOnPaper Can you please also review this PR? I added both Naver and Tumblr here but I can make a separate PR for each of these ones if needed. Thank you,

**Comment by pilcrowonpaper** [comment]
I think I can merge Tumbler but not sure about Naver since there's no English docs

**Comment by mikah13** [comment]
Naver is now removed

File: README.md [hunk 1]
```diff
@@ -36,10 +36,12 @@ const tokens = await github.validateAuthorizationCode(code);
 - Line
 - LinkedIn
 - Microsoft Entra ID
+- Naver
```
Comment (pilcrowonpaper) [inline README.md]:
```suggestion
```

File: src/providers/tumblr.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import { TimeSpan, createDate } from "oslo";
+
+const authorizeEndpoint = "https://www.tumblr.com/oauth2/authorize";
+const tokenEndpoint = "https://api.tumblr.com/v2/oauth2/token";
+
+export class Tumblr implements OAuth2Provider {
+	private client: OAuth2Client;
+	private clientSecret: string;
+
+	constructor(clientId: string, clientSecret: string, redirectURI: string) {
```
Comment (pilcrowonpaper) [inline src/providers/tumblr.ts]: Redirect URI can be optional - see the Github provider source

**Review by mikah13 — COMMENTED** [review]
sorry took me long enough but I have addressed these ones

**Review by pilcrowonpaper — APPROVED** [review]

**Comment by pilcrowonpaper** [comment]
Thanks!

**Pull request merged**